utils/terrascript_client.py

```diff
@@ -435,17 +436,126 @@ def populate_tf_resource_s3(self, resource, namespace_info):
+                            "Principal": {
+                                "Service": "s3.amazonaws.com"
+                            },
+                            "Effect": "Allow",
+                            "Sid": ""
+                        }
+                    ]
+                }
+                rc_values['assume_role_policy'] = role
+                tf_resource = aws_iam_role(id, **rc_values)
+                tf_resources.append(tf_resource)
+
+                # iam policy
+                # Terraform resource reference:
+                # https://www.terraform.io/docs/providers/aws/r/iam_policy.html
+                rc_values.clear()
+                rc_values['name'] = config['rule_name'] + '_iam_policy'
+                policy = {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Action": [
+                                "s3:GetReplicationConfiguration",
+                                "s3:ListBucket"
+                            ],
+                            "Effect": "Allow",
+                            "Resource": [
+                                "${aws_s3_bucket." + identifier + ".arn}"
+                            ]
+                        },
+                        {
+                            "Action": [
+                                "s3:GetObjectVersion",
+                                "s3:GetObjectVersionAcl"
+                            ],
+                            "Effect": "Allow",
+                            "Resource": [
+                                "${aws_s3_bucket." + identifier + ".arn}/*"
+                            ]
+                        },
+                        {
+                            "Action": [
+                                "s3:ReplicateObject",
+                                "s3:ReplicateDelete"
+                            ],
+                            "Effect": "Allow",
+                            "Resource":
+                                "${aws_s3_bucket." +
+                                config['destination_bucket_identifier'] +
+                                ".arn}/*"
+                        }
+                    ]
+                }
+                rc_values['policy'] = policy
+                tf_resource = aws_iam_policy(id, **rc_values)
+                tf_resources.append(tf_resource)
+
+                # iam role policy attachment
+                # Terraform resource reference:
+                # https://www.terraform.io/docs/providers/aws/r/iam_policy_attachment.html
+                rc_values.clear()
+                rc_values['role'] = "${aws_iam_role." + id + ".name}"
+                rc_values['policy_arn'] = "${aws_iam_policy." + id + ".arn}"
+                tf_resource = aws_iam_role_policy_attachment(id, **rc_values)
+                tf_resources.append(tf_resource)
+
+                rc_values.clear()
```

Comment:
can you comment this section with what it does?

Reply:
Done